sdk/attestation/azure-security-attestation/src/attestation_client.cpp

```diff
@@ -215,11 +211,68 @@ void AttestationClient::RetrieveResponseValidationCollateral(
   }
 }
 
-void AttestationClient::CheckAttestationSigners() const
+/** @brief Construct a new Attestation Client object
+ *
+ * @param endpoint The URL address where the client will send the requests to.
+ * @param credential The authentication method to use (required for TPM attestation).
+ * @param options The options to customize the client behavior.
+ */
+AttestationClient const AttestationClient::Create(
```

Comment:
The `not-so-good` about factory methods is that they are accessible from class instances as well.
So, you attestation client will also list the `client.Create()` method just like the class `AttestationClient::Create()`.

What I usually prefer is to create the factory as an additional class with no constructor (real factory). Something like `AttestationClientFactory`.

Reply:
Moved to a separate issue so this commit can be completed.